protocols/directdefund/directdefund_test.go

```diff
@@ -270,7 +270,7 @@ func TestCrankBob(t *testing.T) {
 	}
 
 	// The third crank. Bob is expected to enter the terminal state of the defunding protocol.
-	updated, err = updated.(*Objective).UpdateWithChainEvent(chainservice.AllocationUpdatedEvent{Holdings: types.Funds{}})
```

Comment:
Why was this necessary?

Reply:
I think we are in agreement about the change. I was just wondering about the difference between: `Holdings: types.Funds{}` and `Holdings: types.Funds{types.Address{}: new(big.Int).SetInt64(0)}`. But I think I figured it out -- you've replaced an empty funds mapping with a nonempty mapping with an explicit `0: 0` entry. 

The thrust of this PR implies that `Holdings` must be iterated over: that is, a missing key is no longer the same as a key mapped to `0`. So the test needs to feed in more explicit data (to match the way the chain service will do it). 

Anyway, TLDR: this change makes sense and your suggested change to `AllocationUpdatedEvent` looks good to me.